### PR TITLE
Add CMake build option.

### PR DESCRIPTION
**Pull request by hobbes1069**
I need to package this for Fedora Linux in preparation of OpenColorIO 2.0 release. The simple Makefile is not very packager friendly. This should work for any Linux distro and can be updated for MacOS and Windows if needed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+project(pystring
+    LANGUAGES CXX
+    )
+
+cmake_minimum_required(VERSION 3.0)
+
+set(CMAKE_CXX_FLAGS "-O3 -Wall -Wextra -Wshadow -Wconversion -Wcast-qual -Wformat=2 ${CMAKE_CXX_FLAGS}")
```
Comment (lgritz) [inline CMakeLists.txt:7]: This line feels like it will definitely not work for users trying to build with MSVS.

Comment (hobbes1069) [reply CMakeLists.txt:7]: That's fine, it was left over from the project I copied the CMakeLists.txt file from as a template. Modify as much as you want :)

**Comment by eszlari** [comment]
Why not just copy:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/master/share/cmake/projects/Buildpystring.cmake

**Comment by hobbes1069** [comment]
Mainly because I didn't see it buried 3 directories down. But mine is actually much better. For distro builds shared libraries are preferred and I'd have to fix the hardcoded install locations anyway.

**Comment by grdanny** [comment]
We added a simple cmake configuration.
Thanks for the suggestion!

**Pull request closed without merging**